### PR TITLE
Refactor api comments

### PR DESCRIPTION
**Pull request by ChrisBr**
- Refactor to get rid of unnecessary templates
- Implemented user route
- Implemented specs
- Renamed url parameter to reflect what it actually is

Part of #3618

### CODE REVIEW

File: src/api/spec/controllers/comments_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+RSpec.describe CommentsController, type: :controller do
+  render_views
+  describe 'GET #show' do
+    let(:user) { create(:confirmed_user) }
+
+    RSpec.shared_examples 'request comment index' do
+      it { expect(response).to have_http_status(:success) }
+      it { expect(assigns(:obj)).to eq(comment.commentable) }
+
```
Comment (bgeuken) [inline src/api/spec/controllers/comments_controller_spec.rb]: Nitpick, but would you mind to drop the newline? It seems random here:-)

File: src/api/app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def find_obj
       @obj = User.find_by!(login: params[:user_login])
       @header = { user: @obj.login }
     else
-      @obj = BsRequest.find_by_number!(params[:id])
+      @obj = BsRequest.find_by_number!(params[:request_number])
```
Comment (bgeuken) [inline src/api/app/controllers/comments_controller.rb]: That controller is not in the webui namespace. Is it also exposed to the API?

In that case we would have to keep the old parameter functional. At least for now. We could deprecate it and later drop it.

File: src/api/app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -23,15 +22,18 @@ def destroy
   def find_obj
     if params[:project]
       if params[:package]
-        @template = 'package'
         @obj = Package.get_by_project_and_name(params[:project], params[:package])
+        @header = { project: @obj.project.name, package: @obj.name }
```
Comment (hennevogel) [inline src/api/app/controllers/comments_controller.rb]: `Comment` is polymorphic by now :-)

Comment (ChrisBr) [inline src/api/app/controllers/comments_controller.rb]: Yes, but I'm coming from the parent, the controller needs also to be polymorphic.

File: src/api/app/controllers/comments_controller.rb (hunk 2 above)
Comment (ChrisBr) [inline src/api/app/controllers/comments_controller.rb]: As it is a url parameter it should be fine renaming it:
```
 get 'comments/request/:request_number' => :index, constraints: cons, as: :comments_request
```

Comment (bgeuken) [inline src/api/app/controllers/comments_controller.rb]: Okay. Thanks for checking:+1:

**Comment by ChrisBr** [comment]
[[api] Provide comment API for retrieving comments by author #3618](https://trello.com/c/4JpyirC6/499-api-provide-comment-api-for-retrieving-comments-by-author-3618)

**Pull request merged**